calico_containers/pycalico/util.py

```diff
@@ -104,211 +108,450 @@ def get_hostname():
+
+        try:
+            left_asn_int = int(left_asn)
+            right_asn_int = int(right_asn)
+        except ValueError:
+            raise TypeError("ASDOT notation incorrect. ('{0}' given)")
+
+        if not 0 <= left_asn_int <= 65535:
+            raise ValueError("Left side of as.dot not in range ('{0}' "
+                             "given)".format(left_asn))
+        elif not 0 <= right_asn_int <= 65535:
+            raise ValueError("Right side of as.dot not in range ('{0}' "
+                             "given)".format(right_asn))
+    else:
+        try:
+            asn_int = int(asn)
+        except ValueError:
+            raise TypeError("ASPLAIN number could not be converted to an "
+                            "int. ('{0}' given)".format(asn))
+
+        if not 0 <= asn_int <= 4294967295:
+            raise ValueError("ASPLAIN number not in range ('{0}' given)"
+                             "".format(asn))
 
 
 def validate_characters(input_string):
     """
+    DEPRECATED (use verify_characters)
+
+    Validate that characters in string are supported by Felix.
+    Felix supports letters a-z, numbers 0-9, and symbols _.-
+
+    :param input_string: to be validated
+    :type input_string: str
+    :return: returns True if valid, False if invalid
+    :rtype: bool
+    """
+    return return_bool(verify_characters, input_string)
+
+
+def verify_characters(input_string):
+    """
     Validate that characters in string are supported by Felix.
     Felix supports letters a-z, numbers 0-9, and symbols _.-
 
     :param input_string: string to be validated
-    :return: Boolean: True if valid, False if invalid
+    :type input_string: str
+    :return: None
+    :rtype: None
     """
     # List of valid characters that Felix permits
-    valid_chars = '[a-zA-Z0-9_\.\-]'
+    valid_chars = r'[a-zA-Z0-9_\.\-]'
 
     # Check for invalid characters
     if not re.match("^%s+$" % valid_chars, input_string):
-        return False
-    else:
-        return True
+        raise ValueError("Invalid string. Felix only supports alphanumeric "
+                         "and the symbols '_', '.', and '-' ('{0}' given)"
+                         "".format(input_string))
 
 
-def validate_icmp_type(icmp_type):
+def validate_cidr(cidr):
     """
-    Validate that icmp_type is an integer between 0 and 255.
-    If not return False.
+    DEPRECATED (use verify_cidr)
 
-    :param icmp_type: int value representing an icmp type
-    :return: Boolean: True if valid icmp type, False if not
+    Validate cidr is in correct CIDR notation
+
+    :param cidr: IP addr and associated routing prefix
+    :type cidr: str
+    :return: True if valid IP, False if invalid
+    :rtype: bool
+    """
+    return return_bool(verify_cidr, cidr)
+
+
+def verify_cidr(cidr):
+    """
+    Validate cidr is in correct CIDR notation
+
+    :param cidr: IP addr and associated routing prefix
+    :type cidr: str
+    :return: True if valid IP, False if invalid
```

Comment:
This comment is invalid now, although Github's page doesn't seem to think so. Thought it was my browser caching it, but it shows the same way in others. _shrug_